airflow/providers/amazon/aws/operators/athena.py

```diff
@@ -126,63 +130,97 @@ def execute(self, context: Context) -> str | None:
             )
-        # implicit else:
-        query_status = self.hook.poll_query_status(
-            self.query_execution_id,
-            max_polling_attempts=self.max_polling_attempts,
-            sleep_time=self.sleep_time,
-        )
-
-        if query_status in AthenaHook.FAILURE_STATES:
-            error_message = self.hook.get_state_change_reason(self.query_execution_id)
-            raise AirflowException(
-                f"Final state of Athena job is {query_status}, query_execution_id is "
-                f"{self.query_execution_id}. Error: {error_message}"
+            AthenaQueryResultsLink.persist(
+                context=context,
+                operator=self,
+                region_name=self.hook.conn_region_name,
+                aws_partition=self.hook.conn_partition,
+                query_execution_id=self.query_execution_id,
             )
-        elif not query_status or query_status in AthenaHook.INTERMEDIATE_STATES:
-            raise AirflowException(
-                f"Final state of Athena job is {query_status}. Max tries of poll status exceeded, "
-                f"query_execution_id is {self.query_execution_id}."
+
+            if self.deferrable:
+                self.defer(
+                    trigger=AthenaTrigger(
+                        query_execution_id=self.query_execution_id,
+                        waiter_delay=self.sleep_time,
+                        waiter_max_attempts=self.max_polling_attempts,
+                        aws_conn_id=self.aws_conn_id,
+                        region_name=self.region_name,
+                        verify=self.verify,
+                        botocore_config=self.botocore_config,
+                    ),
+                    kwargs={"query_list": query_list},
+                    method_name="execute_next_query",
+                )
+            # implicit else:
+            query_status = self.hook.poll_query_status(
+                self.query_execution_id,
+                max_polling_attempts=self.max_polling_attempts,
+                sleep_time=self.sleep_time,
             )
 
+            if query_status in AthenaHook.FAILURE_STATES:
+                error_message = self.hook.get_state_change_reason(self.query_execution_id)
+                raise AirflowException(
+                    f"Final state of Athena job is {query_status}, query_execution_id is "
+                    f"{self.query_execution_id}. Error: {error_message}"
+                )
+            elif not query_status or query_status in AthenaHook.INTERMEDIATE_STATES:
+                raise AirflowException(
+                    f"Final state of Athena job is {query_status}. Max tries of poll status exceeded, "
+                    f"query_execution_id is {self.query_execution_id}."
+                )
+
         return self.query_execution_id
 
-    def execute_complete(self, context: Context, event: dict[str, Any] | None = None) -> str:
+    def execute_next_query(
```

Comment:
Instead of running queries one by one and waiting each time each query before running the other I would do it differently. Plus, this is unusual to have task running -> task deferred -> task running task deferred -> etc ... I am not saying this is wrong but I am wondering if it is not anti pattern. Why dont you run all queries and then you wait for all queries to be done? You'd have to update the trigger and the poll logic but this is doable

Reply:
>Why dont you run all queries and then you wait for all queries to be done? 
This feature is for queries that need to be executed sequentially because they depend on each other. For example, one query might drop a temporary table, and the next creates it again.

for example:
```sql
DROP TABLE temporal.transform_step;
CREATE TABLE temporal.transform_step AS SELECT 1;
```
Running them in parallel would cause conflicts. For parallel execution, using TaskFlow with `expand` would be a better approach.

Reply:
I see

Reply:
Can this be mentioned somewhere as note/in doc? so user will get to know the real use of sequential here.